packages/admin-next/dashboard/src/routes/orders/order-edit/components/order-edit-form/order-edit-form.tsx

```diff
@@ -0,0 +1,354 @@
+
+type OrderEditFormProps = {
+  order: Order
+  orderEdit: OrderEdit
+}
+
+const QuantitiesSchema = zod.union(
+  zod.record(zod.string(), zod.number().optional()),
+  zod.object({ note: zod.string().optional() })
+)
+
+export function OrderEditForm({ order, orderEdit }: OrderEditFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+  const [, setSearchParams] = useSearchParams()
+
+  /**
+   * STATE
+   */
+  const [open, setOpen] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+
+  /**
+   * FORM
+   */
+  const form = useForm<zod.infer<typeof QuantitiesSchema>>({
+    defaultValues: order.items.reduce(
+      (acc, i) => {
+        acc[i.id] = i.quantity
+        return acc
+      },
+      { note: orderEdit.internal_note || "" }
+    ),
+  })
+
+  /**
+   * CRUD HOOKS
+   */
+  const { mutateAsync: confirmOrderEdit } = useAdminConfirmOrderEdit(
+    orderEdit.id
+  )
+  const { mutateAsync: cancelOrderEdit } = useAdminCancelOrderEdit(orderEdit.id)
+  const { mutateAsync: addLineItemToOrderEdit } = useAdminOrderEditAddLineItem(
+    orderEdit.id
+  )
+  const { mutateAsync: updateOrderEdit } = useAdminUpdateOrderEdit(orderEdit.id)
+
+  const onQuantityChangeComplete = async (itemId: string) => {
+    setIsLoading(true)
+
+    const quantity = form.getValues()[itemId]
+
+    // if (form.getValues()[itemId] === 0) {
+    //   await medusa.admin.orderEdits.removeLineItem(orderEdit.id, itemId)
+    // } else
+
+    if (quantity > 0) {
+      await medusa.admin.orderEdits.updateLineItem(orderEdit.id, itemId, {
+        quantity,
+      })
+    }
+    await queryClient.invalidateQueries(
+      adminOrderEditsKeys.detail(orderEdit.id)
+    )
+
+    setIsLoading(false)
+  }
+
+  const currentItems = useMemo(
+    () =>
+      orderEdit?.items
+        .sort((i1, i2) => i1.id.localeCompare(i2.id))
+        .filter((i) => i.original_item_id) || [],
+    [orderEdit]
+  )
+  const addedItems = useMemo(
+    () =>
+      orderEdit?.items
+        .sort((i1, i2) => i1.id.localeCompare(i2.id))
+        .filter((i) => !i.original_item_id) || [],
+    [orderEdit]
+  )
+
+  /**
+   * EFFECTS
+   */
+  useEffect(() => {
+    if (orderEdit) {
+      orderEdit?.items.forEach((i) => {
+        form.setValue(i.id, i.quantity)
+      })
+    }
+  }, [orderEdit?.items.length])
+
+  /**
+   * HANDLERS
+   */
+  const handleOpenChange = (open: boolean) => {
+    if (!open) {
+      setSearchParams(
+        {},
+        {
+          replace: true,
+        }
+      )
+    }
+
+    setOpen(open)
+  }
+
+  const onVariantsSelect = async (variantIds: string[]) => {
+    setIsLoading(true)
+    try {
+      await Promise.all(
+        variantIds.map((id) =>
+          addLineItemToOrderEdit({ variant_id: id, quantity: 1 })
+        )
+      )
+
+      await queryClient.invalidateQueries(adminOrderKeys.detail(order.id))
+    } finally {
+      setIsLoading(false)
+    }
+
+    setOpen(false)
+  }
```

Comment:
This is displaying the spinner in the wrong button (the submit button instead of the add button)